src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -2864,11 +2877,13 @@ void ClientChannel::LoadBalancedCall::RecvTrailingMetadataReady(
     }
     // If the LB policy requested a callback for trailing metadata, invoke
     // the callback.
-    if (self->lb_recv_trailing_metadata_ready_ != nullptr) {
+    if (self->lb_subchannel_call_tracker_ != nullptr) {
       Metadata trailing_metadata(self, self->recv_trailing_metadata_);
-      LbCallState lb_call_state(self);
-      self->lb_recv_trailing_metadata_ready_(status, &trailing_metadata,
-                                             &lb_call_state);
+      BackendMetricAccessor backend_metric_accessor(self);
+      LoadBalancingPolicy::SubchannelCallTrackerInterface::FinishArgs args = {
+          status, &trailing_metadata, &backend_metric_accessor};
+      self->lb_subchannel_call_tracker_->Finish(args);
+      self->lb_subchannel_call_tracker_.reset();
```

Comment:
Is there any reason to not leave this to the destructor?

Reply:
Seems more optimal to do it here.  The destructor could be delayed a bit waiting for various callbacks, but we know that we don't need this object anymore after this point.